models/sessions/scratch/snowplow_unified_sessions_this_run.sql

```diff
@@ -233,14 +233,17 @@ with session_firsts as (
         {%- for conv_def in var('snowplow__conversion_events') %}
             {{ snowplow_unified.get_conversion_columns(conv_def)}}
         {%- endfor %}
-    from {{ ref('snowplow_unified_events_this_run') }}
-    where
-        1 = 1
-        {% if var("snowplow__ua_bot_filter", true) %}
+        {% if var('snowplow__enable_conversions', false) %}
+          from {{ ref('snowplow_unified_conversions_this_run') }}
+          group by session_identifier
+        {% else %}
+          from {{ ref('snowplow_unified_events_this_run') }}
+          where 1 = 1
+          {% if var("snowplow__ua_bot_filter", true) %}
             {{ filter_bots() }}
+          {% endif %}
+          group by session_identifier
         {% endif %}
```

Comment:
Not that important but you could stick the group by outside the if given it's in both